oxia-operator/controllers/oxiacluster_coordinator.go

```diff
@@ -0,0 +1,218 @@
+		if err != nil {
+			r.Log.Error(err, "Failed to create service account :", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Created service account: ", "NamespaceName", nsName)
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if oxia.DeletionTimestamp != nil {
+		r.Log.Info("Service account deletion is unsupported yet!")
+	}
+
+	// Reconcile role
+	_role := &rbacV1.Role{}
+	nsName = types.NamespacedName{Name: resourceName(Coordinator, oxia.Name), Namespace: oxia.Namespace}
+	err = r.Client.Get(ctx, nsName, _role)
+	if err != nil {
+		if !apiErrors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to get role: ", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Creating a new role :", "NamespaceName", nsName)
+		err := r.Client.Create(ctx, role(Coordinator, oxia))
+		if err != nil {
+			r.Log.Error(err, "Failed to create role:", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Created role: ", "NamespaceName", nsName)
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if oxia.DeletionTimestamp != nil {
+		r.Log.Info("Role deletion is unsupported yet!")
+	}
+
+	// Reconcile role binding
+	_roleBinding := &rbacV1.RoleBinding{}
+	nsName = types.NamespacedName{Name: resourceName(Coordinator, oxia.Name), Namespace: oxia.Namespace}
+	err = r.Client.Get(ctx, nsName, _roleBinding)
+	if err != nil {
+		if !apiErrors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to get role binding: ", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Creating a new role binding:", "NamespaceName", nsName)
+		err := r.Client.Create(ctx, roleBinding(Coordinator, oxia))
+		if err != nil {
+			r.Log.Error(err, "Failed to create role binding:", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Created role binding: ", "NamespaceName", nsName)
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if oxia.DeletionTimestamp != nil {
+		r.Log.Info("Role binding deletion is unsupported yet!")
+	}
+
+	// reconcile configmap
+	_configMap := &coreV1.ConfigMap{}
+	nsName = types.NamespacedName{Name: resourceName(Coordinator, oxia.Name), Namespace: oxia.Namespace}
+	err = r.Client.Get(ctx, nsName, _configMap)
+	if err != nil {
+		if !apiErrors.IsNotFound(err) {
+			r.Log.Error(err, "Failed to get configmap: ", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Creating a new configmap:", "NamespaceName", nsName)
+		configMap, err := coordinatorConfigMap(oxia)
+		if err != nil {
+			r.Log.Error(err, "Failed to create configmap:", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		err = r.Client.Create(ctx, configMap)
+		if err != nil {
+			r.Log.Error(err, "Failed to create configmap:", "NamespaceName", nsName)
+			return ctrl.Result{}, err
+		}
+		r.Log.Info("Created configmap: ", "NamespaceName", nsName)
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	if oxia.DeletionTimestamp != nil {
```

Comment:
Does it make sense that if resources are configured with ownerReference, these resources will be automatically cleaned up